### ISSUE TITLE
Metaphase Spindle Analysis 

### ISSUE THREAD

**Issue by RankAma** [embl-cba/spindle3d#33]
I've recently captured images of U2OS cells undergoing mitosis following RNAi screen and wanted analyse the effect on spindles.  The image meets the input data requirements as per the Spindle3D Read me (2 colour confocal image stack) .
 
![metaphase spindle 2](https://user-images.githubusercontent.com/83732729/150387925-c16682bb-5c33-49a9-8018-a61af3db5ae7.jpg)
![metaphase spindle](https://user-images.githubusercontent.com/83732729/150388348-7395aa05-506c-4ffa-bba5-b29aacaa5936.jpg)

Unfortunately, when I run the plugin in Fiji it fails to carryout spindle measurements and given the following message:

## Spindle3D Measurements
Processing image: Stack-1-1
Running Spindle3D v0.8.0
Create isotropic images...
DNA downscaled minimum value: 0.0
DNA downscaled maximum value: 223.0
DNA initial threshold factor: 0.5
DNA initial threshold = ( max - min ) * factor + min: 111.5
Removed 0 of 2 regions, because of image border contact.
Determining DNA axes...
Dna ellipsoid fit shortest axis length: 6.953510001605706
Dna ellipsoid fit longest axis length: 136.10308678177435
Measuring DNA width...

## Spindle3D Measurements
{
  "version": "Spindle3D version: 0.8.0",
  "dnaThreshold": NaN,
  "metaphasePlateLength": NaN,
  "metaphasePlateWidth": NaN,
  "chromatinVolume": NaN,
  "chromatinDilation": NaN,
  "spindleLength": NaN,
  "spindleThreshold": NaN,
  "spindleSNR": NaN,
  "spindleVolume": NaN,
  "spindleWidthMin": NaN,
  "spindleWidthMax": NaN,
  "spindleCenterToMetaphasePlateCenterDistance": NaN,
  "spindleAngle": NaN,
  "tubulinSpindleIntensityVariation": NaN,
  "tubulinSpindleAverageIntensity": NaN,
  "tubulinCellularAverageIntensity": NaN,
  "tubulinCytoplasmAverageIntensity": NaN,
  "spindleWidthAvg": NaN,
  "spindleAspectRatio": NaN,
  "cellVolume": NaN,
  "cellSurface": NaN,
  "log": "Exception during computation: \njava.lang.ArrayIndexOutOfBoundsException"
}
 
Exception during computation: 
java.lang.ArrayIndexOutOfBoundsException

Sometimes it states the chromatin is touching the border when that isnt the case.
Is there a reason why this is keeps happening and could anyone provide a solution for this  ?

**Comment by tischi** [comment]
@RankAma 
Thanks a lot for reporting this!
Could you please provide the raw data for download?
I could try to test what's happening.

**Comment by tischi** [comment]
In fact, in the image that you provided the chromatin seems to be touching the image boundary:

<img width="310" alt="image" src="https://user-images.githubusercontent.com/2157566/150396427-8434eb93-2ead-4d4d-974b-9671eb93e27a.png">

We are at the lowest z-plane and we can still see the chromatin, thus it is probably "touching the bottom" of the image. The point being that we do the analysis fully in 3D and thus both the spindle and chromatin also must be fully in the image along the z-direction.

But, as said, if you could provide some raw data we could explore more (also ping @TobiasKletter).

**Comment by RankAma** [comment]
Dear tischi,

Thank you for getting back to me so quickly. Here is my raw data, unfortunately Github doesnt support TIF so i sent my files as a JPEG. I removed the red channel as the staining was inefficient so i focused on Tubulin staining (green) and chromatin with DAPI/Hoechst(blue) .   

The raw data shows multiple cells, I stacked the z planes in 2 colour channels and cropped an individual cell with a clear metaphase spindle. So the chromatin you're referring to is the blue staining at the bottom left of the image. @TobiasKletter, i've tried to follow the steps youre used in your paper (a nice read), please let me know if there's anything you recommend i do.
 
![Index idx xml  Well 2, Field 1 _c00_z01_t01](https://user-images.githubusercontent.com/83732729/150401343-4c0d6859-51ea-462e-b689-3eca92cccb8c.jpg)
![Index idx xml  Well 2, Field 1 _c00_z02_t01](https://user-images.githubusercontent.com/83732729/150401347-b8f10b3d-08c3-480e-b2a0-426f6b502448.jpg)
![Index idx xml  Well 2, Field 1 _c00_z03_t01](https://user-images.githubusercontent.com/83732729/150401348-97e8fb13-d485-4db2-80d0-7c56ffcb7138.jpg)
![Index idx xml  Well 2, Field 1 _c00_z04_t01](https://user-images.githubusercontent.com/83732729/150401349-54061997-ae36-4ffd-bc57-ef26120acf62.jpg)
![Index idx xml  Well 2, Field 1 _c00_z05_t01](https://user-images.githubusercontent.com/83732729/150401351-babdfb92-b928-4651-9694-5a3168263f68.jpg)
![Index idx xml  Well 2, Field 1 _c00_z06_t01](https://user-images.githubusercontent.com/83732729/150401353-c156cc5f-9928-4bfc-870c-7cecdeadeaf3.jpg)
![Index idx xml  Well 2, Field 1 _c01_z01_t01](https://user-images.githubusercontent.com/83732729/150401356-a0617cef-9140-40a6-a436-f092573151e4.jpg)
![Index idx xml  Well 2, Field 1 _c01_z02_t01](https://user-images.githubusercontent.com/83732729/150401359-175936e4-f339-4b5f-9065-8c160b38a453.jpg)
![Index idx xml  Well 2, Field 1 _c01_z03_t01](https://user-images.githubusercontent.com/83732729/150401360-b3400a24-d55c-4f06-a8d5-b759768652c2.jpg)
![Index idx xml  Well 2, Field 1 _c01_z04_t01](https://user-images.githubusercontent.com/83732729/150401361-bef0e39c-6079-4661-a033-cb33c5e8a663.jpg)
![Index idx xml  Well 2, Field 1 _c01_z05_t01](https://user-images.githubusercontent.com/83732729/150401364-52ec1957-1bec-4cf6-bf84-70fccb4f7f27.jpg)
![Index idx xml  Well 2, Field 1 _c01_z06_t01](https://user-images.githubusercontent.com/83732729/150401368-d3be9219-aa4f-47f2-9167-5b29eccd5e23.jpg)
![Index idx xml  Well 2, Field 1 _c02_z01_t01](https://user-images.githubusercontent.com/83732729/150401372-25d419bb-2f65-457a-b1db-88ac4f375310.jpg)
![Index idx xml  Well 2, Field 1 _c02_z02_t01](https://user-images.githubusercontent.com/83732729/150401374-ec7ae7e4-82ed-4833-ae2c-aca4b8def800.jpg)
![Index idx xml  Well 2, Field 1 _c02_z03_t01](https://user-images.githubusercontent.com/83732729/150401377-b1c2d489-c420-4cba-8d68-f4fafbc0f01a.jpg)
![Index idx xml  Well 2, Field 1 _c02_z04_t01](https://user-images.githubusercontent.com/83732729/150401379-a11bf4fc-90da-41c6-8447-8152681d35ad.jpg)
![Index idx xml  Well 2, Field 1 _c02_z05_t01](https://user-images.githubusercontent.com/83732729/150401380-7abb9fc6-6be5-4a47-a177-d7f27dbcdff1.jpg)
![Index idx xml  Well 2, Field 1 _c02_z06_t01](https://user-images.githubusercontent.com/83732729/150401382-12c685de-4f9b-4a0d-b6d6-bfb34f08b33a.jpg)
[Batch_Image_Export.txt](https://github.com/embl-cba/spindle3d/files/7907710/Batch_Image_Export.txt)

**Comment by tischi** [comment]
Hi @RankAma,

Thanks a lot, but we would really need the raw data.
Could you please try to create a download link for the TIFF stack, e.g. pointing to a dropbox or something similar?

**Comment by RankAma** [comment]
Hey Tischi,

Yh I've created a public folder on Google drive with the raw data z-stack with all channels titled 'Stack_all_channels' (including red channel which i dont want). I made another stack of the same image with composite of different channels titled 'Stacked_all_channels_composite' and a stack tif of the first image titled cropped2channel_stack-1-1 . Please let me know if there's anything else you need, or if the data needs to be uploaded in another format.

https://drive.google.com/drive/folders/1DpX1m7evmx9nof7kw-7dzfPFTzYwFQWL?usp=sharing

**Comment by TobiasKletter** [comment]
Hi @RankAma ,

Thanks for sharing your problem and the images!

There are two main issues: Please make sure the voxel size in your image is correctly annotated. In the cropped file that you sent us, this is not the case: 
![image](https://user-images.githubusercontent.com/45814595/150528330-fae54af1-e768-4358-ac6b-b6a85369fab1.png)

Please check your meta data in the raw microscope files for this information and then copy them to the image properties via FIJI > Image > Properties... Alternatively, just crop your cell of interest from the raw image by drawing a rectangular selection and go Image > Duplicate to maintain the original calibration.

Secondly, as Tischi mentioned above, the Spindle3D analysis will abort when the segmented objects (chromatin plate or spindle) are directly contacting any of the image borders. In your particular case, both the spindle and the chromatin are touching the "floor" of your image. 
You can trick the analysis by adding empty slices at the beginning or end of your stack. (Image > Stacks > Add slice; in your case choose "prepend slice") 
However, your morphometry results will then of course be incorrect, since you are missing parts of the objects. I am sorry, but the best advice that I can give you in this case is to acquire new images that cover the full extents of your cells in Z. You can check ideal input images for the plugin via Plugins > Spindle3D > Download Example Images.

Please don't hesitate to ask if anything remains unclear! :)

**Comment by RankAma** [comment]
Hi Tobias,

Thank you very much for your detailed response, i've been trying to apply the same dimensions to Fiji:

Acquisition Date:	2021-11-04 19:22:03
Import Date:	2021-11-05 09:38:49
Dimension (XY):	2160 x 2160
Pixels Type:	uint16
Pixels Size (XYZ) (µm):	0.15 x 0.15 x 0.00
Z-sections:	6
Timepoints:	1
Channels:	DAPI, Alexa 488, Alexa 568

I was unable to change from 8-bit to 16-bit type image but ran the Spindle3D again.The good news is i got a few spindle measurements. However, the program states that all regions DNA chromatin were touching the border (by border does this mean the end of the image). This might be a stupid question, but what does it mean by border ? My understanding is that this would be the boundaries of the image. The chromatin in the previous image is clearly in the middle so i'm confused when you say the chromatin is in the touching the borders. Can you please clarify this for me ?

 I compared the orthogonal views of my image (Operetta_40x_U2OS_siNcon-1) to the sample image NikonSD_60x_HEK293SiRTubulin_01, but i dont know how important the differences are between the two or if i'm not meeting image processing requirements. Operetta_40x_U2OS_siNcon-1 image is in the G-drive. 

![image](https://user-images.githubusercontent.com/83732729/150575942-ab007a3a-5e5d-4cfd-81f0-78425e4e89a7.png)
![image](https://user-images.githubusercontent.com/83732729/150576182-af59cecf-bea7-4385-86d9-c01c278abe5c.png)

**Comment by TobiasKletter** [comment]
Hi @RankAma 


> Pixels Size (XYZ) (µm): 0.15 x 0.15 x 0.00 
> Z-sections: 6

This part is critical: While we now know the X and Y pixel lengths (0.15 µm x 0.15 µm), we still lack the Z "height" information, that is the _distance between the Z-sections_. 
This is also the reason why your sample looks so flat in the orthogonal views! Fiji thinks there are 6 slices with Zero thickness, so they are displayed as a super flat z-stack. For illustrative purposes, I entered the following information (I basically assumed a z-step size of 0.75 µm) and got the following orthogonal views:

![image](https://user-images.githubusercontent.com/45814595/150789487-e4d23e33-c079-4722-9f14-ccfbc1e036e4.png)
![image](https://user-images.githubusercontent.com/45814595/150790231-11fc6180-b74c-485d-87e6-a59dbc94115c.png)

I highlighted all image borders with an orange dashed line. While in the XY plane you're safe, your sample nevertheless is directly contacting the line in the orthogonal views. The plugin will recognise this and abort...

This is not the case in the example image, neither in the XY plane nor in the orthogonal views. Essentially, there is "empty" space around AND above and below your spindle and chromatin:

 ![image](https://user-images.githubusercontent.com/83732729/150576182-af59cecf-bea7-4385-86d9-c01c278abe5c.png)

I hope this illustrates it?
To sum up, quite generally you want to figure out what your z-step distance is and insert that information to the image properties. 
And more specifically for your image: if you want to quantify spindle and chromatin morphologies in 3D, please take care to fully cover the entire Z depth of your objects of interest. 

Good luck!

PS:
> I was unable to change from 8-bit to 16-bit type image but ran the Spindle3D again.

No need to convert your input bit depth! The plugin works on both 8-bit and 16-bit input, preferably don't change that from the original microscope data.

**Comment by tischi** [comment]
@TobiasKletter Thanks a lot for the detailed explanation!
Do you think you could add this to the README?
Maybe only showing the Orthogonal Views of our sample data where there is enough space around the cell in 3D.

Unless @RankAma would kindly allow us to use his data for the README to illustrate the "touching the image border at the top and bottom" issue?

**Comment by RankAma** [comment]
Hey @tischi and @TobiasKletter,

That's fine, It's Ok for you to use my data for the README.

I've directly imported the raw data with the Z- height information directly into Fiji and the two stack colour image is still touching the border. I dont know if this because i'm cropping the raw image to focus on a single cell or something else. I've added the uncropped and cropped version to in folder 'Operetta_40x_U2OS_siNcon-2'.

![image](https://user-images.githubusercontent.com/83732729/150834615-2cf742e6-7abc-404d-a969-ae355c86dc09.png)


Is possible it to use analyse my spindles with the data i've provided or is it not possible with the images ?

**Comment by TobiasKletter** [comment]
@tischi 
Will do!

@RankAma 
Thanks for sharing your images! 
I checked your recent uploads and the pixel size annotation seems fine now! 👍 

However, the main problem persists and Spindle3D will always abort the analysis because of your limited Z range. Even if it wouldn't abort, the 3D measurements would not be meaningful. Your images otherwise look really good and might be absolutely suitable for whatever you wanted to look at but I am afraid for your images to be analysable with the plugin you would need to go back to the microscope again, sorry! 
A good starting point that I would suggest is to find the central plane of your spindle and record a Z range of at least 30 µm (that is, 15 µm plus and 15 µm minus your central position), that should cover the entire mitotic cell.
